### PR TITLE
Symbol query fixes

### LINKED ISSUE #250: When querying using multiple labels, attributes assignment causes Cypher error
This becomes an issue for example when[ using subclasses in neo4j](https://github.com/neo4jrb/neo4j/issues/1133).

### PR DESCRIPTION
**Pull request by brucek**
Fixes #250, plus a couple other minor things

### CODE REVIEW

File: spec/neo4j-core/unit/query_spec.rb [hunk 1]
```diff
@@ -639,6 +639,10 @@ def self.it_generates(cypher, params = {})
     describe '.create(q: {Person: {age: nil, height: 70}})' do
       it_generates 'CREATE (q:`Person` {age: {q_Person_age}, height: {q_Person_height}})', q_Person_age: nil, q_Person_height: 70
     end
+
+    describe '.create(q: {\'Child:Person\': {age: 41, height: 70}})' do
+      it_generates 'CREATE (q:`Child:Person` {age: {q_Child_Person_age}, height: {q_Child_Person_height}})', q_Child_Person_age: 41, q_Child_Person_height: 70
```
Comment (subvertallchris) [inline spec/neo4j-core/unit/query_spec.rb]: Just to be sure, did you check that this creates `Child` and `Person` as two separate labels and not some mutant `Child:Person`?

Comment (subvertallchris) [inline spec/neo4j-core/unit/query_spec.rb]: You might also want to test `.create('Child:Person': {age: 41, height: 70})`, without the node identifier.

**Comment by brucek** [comment]
You were right, that created the mutant `Child:Person`.  I pushed something that works, but I'm not 100% sure if we would ever see `::` as a `label_arg` to `format_label` for any reason that should be preserved.  If not, this should now work I think.

**Comment by brucek** [comment]
...or not.  Guess I need to get JRuby running now :-/

**Comment by subvertallchris** [comment]
It'll chill out if you use a hash rocket instead of `{"Key": val}` It doesn't know how to symbolize the key implicitly.

File: spec/neo4j-core/unit/query_spec.rb [hunk 2]
```diff
@@ -639,6 +639,14 @@ def self.it_generates(cypher, params = {})
...
       it_generates 'CREATE (q:`Person` {age: {q_Person_age}, height: {q_Person_height}})', q_Person_age: nil, q_Person_height: 70
     end
+
+    describe ".create(q: {:'Child:Person' => {age: 41, height: 70}})" do
+      it_generates 'CREATE (q:`Child`:`Person` {age: {q_Child_Person_age}, height: {q_Child_Person_height}})', q_Child_Person_age: 41, q_Child_Person_height: 70
+    end
+
+    describe ".create(:'Child:Person' => {age: 41, height: 70})" do
+      it_generates 'CREATE (:`Child`:`Person` {age: {Child_Person_age}, height: {Child_Person_height}})', Child_Person_age: 41, Child_Person_height: 70
+    end
```
Comment (cheerfulstoic) [inline spec/neo4j-core/unit/query_spec.rb]: I'm torn.  I feel like if you specify a key of `:'Child:Person'` that you it should be specifying a label which is `Child:Person`.  It is legal for a label to have a colon in it.

One way that would perhaps be better is to specify an `Array` as the key like this:

`.create([:Child, :Person] => {age: 41, height: 70})`

That could even be:

`.create(%i(Child Person) => {age: 41, height: 70})`

Thoughts?

Comment (brucek) [inline spec/neo4j-core/unit/query_spec.rb]: I'm actually trying to fix neo4jrb/neo4j/issues/1133, and I'll definitely defer to you on this (makes sense to me).

I think this means that this line is where my problem arises:  https://github.com/neo4jrb/neo4j/blob/master/lib/neo4j/shared/query_factory.rb#L72 as the .to_sym is combining all the labels into a single sym then passed into here.

...and that the array solution makes it straightforward to fix there too.  Let me know if that all sounds right, thanks Brian

Comment (brucek) [inline spec/neo4j-core/unit/query_spec.rb]: OK, I think what I did supports your idea of an array of sym's.  The %i version didn't work out of the box and it seems like the array of sym's is an improvement at least and I didn't want to dig further without a clear need - hope that's ok :-)

**Comment by brucek** [comment]
I think at the end of the day I just changed the `attributes_string` to strip out all colons(:), some of those tests are probably unneeded so I can take them out if they seem like overkill - they might be helpful as documentation so I just left them for now.

**Comment by cheerfulstoic** [comment]
Getting back after some business.  This looks great, thanks!

**Comment by cheerfulstoic** [comment]
Busy-ness, that is ;)

**Comment by cheerfulstoic** [comment]
FYI on a small change that I made to this: https://github.com/neo4jrb/neo4j-core/commit/ccffee4a183d680dc9d76f3f86ca9c384e3df5d9

**Pull request merged**